backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/impl/PPRoutingRepository.java

```diff
@@ -130,7 +135,13 @@ private PPRouting toRouting(final I_AD_Workflow routingRecord)
 				.map(productRecord -> toRoutingProduct(productRecord, nodeIdToActivityIds))
 				.collect(ImmutableList.toImmutableList());
 
-		final PPRoutingActivityId firstActivityId = PPRoutingActivityId.ofAD_WF_Node_ID(routingId, routingRecord.getAD_WF_Node_ID());
+		final int wfNodeId = routingRecord.getAD_WF_Node_ID();
+
+		if(wfNodeId <= 0)
+		{
+			throw new AdempiereException(MSG_AD_Workflow_Missing_Node, routingRecord);
```

Comment:
instead of routingRecord i think only routing's name would be much more use friendly